### PR TITLE
fix: wrap fields in types in quotes

### LINKED ISSUE #5973: CAST from null to STRUCT doesn't preserve case for STRUCT fields
**Describe the bug**
When CASTING NULL to a struct with case sensitive fields, the fields all become capitalized instead of having the case preserved.

**To Reproduce**
Start up ksqlDB server with Schema Registry enabled
```
CREATE STREAM `input_test_stream` ( 
   `field` VARCHAR 
 ) WITH ( 
   KAFKA_TOPIC='input', Partitions=2,
   VALUE_FORMAT='AVRO' 
 ); 
 
 CREATE STREAM `output_test_stream` ( 
   `body` STRUCT< `parent` ARRAY<STRUCT< `caseSensitiveChildren` VARCHAR >> > 
  ) WITH ( 
    KAFKA_TOPIC='output', Partitions=2,
    VALUE_FORMAT='AVRO' 
  ); 

  INSERT INTO `output_test_stream` 
    SELECT STRUCT( 
      `parent` := CAST("" AS ARRAY<STRUCT<`caseSensitiveChildren` VARCHAR>>) 
    ) AS 
     `body` 
     FROM `input_test_stream`;
```

Another case:
```
CREATE STREAM `input_test_stream_2` ( 
   `anything` VARCHAR 
 ) WITH ( 
   KAFKA_TOPIC='input_topic', Partitions=2,
   VALUE_FORMAT='Delimited' 
 ); 

CREATE STREAM output_query WITH (VALUE_FORMAT='AVRO') as select STRUCT( 
      `testField` := CAST(NULL AS ARRAY<STRUCT<`caseSensitiveField` VARCHAR>>) 
    ) from `input_test_stream_2`;

// produce data to topic 'input_topic'
```

**Expected behavior**
The schema for the output topics should still have case sensitive fields

**Actual behaviour**
The schema has changed, the field is now all capitalized
```
curl -X GET http://localhost:8081/subjects/OUTPUT_QUERY-value/versions/1
{"subject":"OUTPUT_QUERY-value","version":1,"id":33,"schema":"{\"type\":\"record\",\"name\":\"KsqlDataSourceSchema\",\"namespace\":\"io.confluent.ksql.avro_schemas\",\"fields\":[{\"name\":\"KSQL_COL_0\",\"type\":[\"null\",{\"type\":\"record\",\"name\":\"KsqlDataSourceSchema_KSQL_COL_0\",\"fields\":[{\"name\":\"testField\",\"type\":[\"null\",{\"type\":\"array\",\"items\":[\"null\",{\"type\":\"record\",\"name\":\"KsqlDataSourceSchema_KSQL_COL_0_testField\",\"fields\":[{\"name\":\"caseSensitiveField\",\"type\":[\"null\",\"string\"],\"default\":null}],\"connect.name\":\"io.confluent.ksql.avro_schemas.KsqlDataSourceSchema_KSQL_COL_0_testField\"}]}],\"default\":null}],\"connect.name\":\"io.confluent.ksql.avro_schemas.KsqlDataSourceSchema_KSQL_COL_0\"}],\"default\":null}],\"connect.name\":\"io.confluent.ksql.avro_schemas.KsqlDataSourceSchema\"}"}

curl -X GET http://localhost:8081/subjects/OUTPUT_QUERY-value/versions/2
{"subject":"OUTPUT_QUERY-value","version":2,"id":34,"schema":"{\"type\":\"record\",\"name\":\"KsqlDataSourceSchema\",\"namespace\":\"io.confluent.ksql.avro_schemas\",\"fields\":[{\"name\":\"KSQL_COL_0\",\"type\":[\"null\",{\"type\":\"record\",\"name\":\"KsqlDataSourceSchema_KSQL_COL_0\",\"fields\":[{\"name\":\"testField\",\"type\":[\"null\",{\"type\":\"array\",\"items\":[\"null\",{\"type\":\"record\",\"name\":\"KsqlDataSourceSchema_KSQL_COL_0_testField\",\"fields\":[{\"name\":\"CASESENSITIVEFIELD\",\"type\":[\"null\",\"string\"],\"default\":null}]}]}],\"default\":null}]}],\"default\":null}]}"
```

**Additional context**
I think the cause is that the quotes are missing from the SelectExpression in the QueryPlan for the corresponding command in the command topic

```
"selectExpressions":["STRUCT(`testField`:=CAST(null AS ARRAY<STRUCT<caseSensitiveField STRING>>)) AS KSQL_COL_0"]}
```

### PR DESCRIPTION
**Pull request by jzaralim**
### Description 

Previously, types in expressions were not quoted in the QueryPlan, causing fields in STRUCTS to not have their casing preserved. This change just moves the formatting logic from `ExpressionFormatter` to the subtypes of `SqlType`, so reserved words like `STRUCT` and `MAP` remain unquoted.

Fixes #5973 

edit: Now everything in expressions are quoted, most notably column names. The updated unit tests reflect these changes.

### Testing done 
Updated unit tests, manually tested

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

edit: apologies for the massive pr, had to update qtt query plans. ~~there are really only 2 lines changed under `ExpressionFormatter.java` and `ExpressionFormatterTest.java`~~

### CODE REVIEW

File: ksqldb-execution/src/main/java/io/confluent/ksql/execution/expression/formatter/ExpressionFormatter.java [hunk 1]
```diff
@@ -90,7 +90,7 @@ private Context(final FormatOptions formatOptions) {
 
     @Override
     public String visitType(final Type node, final Context context) {
-      return node.getSqlType().toString(context.formatOptions);
+      return node.getSqlType().toString();
```
[the remark this answers is not in the document]
Comment (agavra) [reply ksqldb-execution/src/main/java/io/confluent/ksql/execution/expression/formatter/ExpressionFormatter.java:93]: Looking into the code, I'm not sure why this is the default format options:
```java
  public static String formatExpression(final Expression expression) {
    return formatExpression(expression, FormatOptions.of(s -> false));
  }
```
I think we should probably default to escaping everything, not nothing - that might be the problem but I haven't dug into it at all

[the remark this answers is not in the document]
Comment (jzaralim) [reply ksqldb-execution/src/main/java/io/confluent/ksql/execution/expression/formatter/ExpressionFormatter.java:93]: It does work, but it also escapes **everything** - column names, function names etc. I tested manually, and it doesn't seem to break anything. The only difference I found was that error messages have quotes everywhere.

**Comment by jzaralim** [comment]
closed because this branch got too messy. will open another PR.

**Comment by jzaralim** [comment]
New PR here: #6245

**Pull request closed without merging**